diff -up plasma-workspace-6.3.3/appmenu/appmenu.cpp.omv~ plasma-workspace-6.3.3/appmenu/appmenu.cpp
--- plasma-workspace-6.3.3/appmenu/appmenu.cpp.omv~	2025-03-14 02:26:46.553487615 +0100
+++ plasma-workspace-6.3.3/appmenu/appmenu.cpp	2025-03-14 02:27:17.760379566 +0100
@@ -98,7 +98,7 @@ AppMenuModule::AppMenuModule(QObject *pa
         m_xcbConn = xcb_connect(nullptr, nullptr);
     }
 #endif
-    if (qGuiApp->platformName() == QLatin1String("wayland")) {
+    if (qGuiApp->platformName().startsWith(QLatin1String("wayland"))) {
         auto connection = KWayland::Client::ConnectionThread::fromApplication();
         KWayland::Client::Registry registry;
         registry.create(connection);
