diff -up plasma-workspace-6.3.0/startkde/startplasma-wayland.cpp.2~ plasma-workspace-6.3.0/startkde/startplasma-wayland.cpp
--- plasma-workspace-6.3.0/startkde/startplasma-wayland.cpp.2~	2025-02-06 12:03:03.000000000 +0100
+++ plasma-workspace-6.3.0/startkde/startplasma-wayland.cpp	2025-02-12 02:18:27.752534435 +0100
@@ -22,6 +22,7 @@ int main(int argc, char **argv)
 
     // Let clients try to reconnect to kwin after a restart
     qputenv("QT_WAYLAND_RECONNECT", "1");
+    qputenv("QT_QPA_PLATFORM", "wayland");
 
     // Query whether org.freedesktop.locale1 is available. If it is, try to
     // set XKB_DEFAULT_{MODEL,LAYOUT,VARIANT,OPTIONS} accordingly.
diff -up plasma-workspace-6.3.0/startkde/startplasma-x11.cpp.2~ plasma-workspace-6.3.0/startkde/startplasma-x11.cpp
--- plasma-workspace-6.3.0/startkde/startplasma-x11.cpp.2~	2025-02-12 02:18:27.752697982 +0100
+++ plasma-workspace-6.3.0/startkde/startplasma-x11.cpp	2025-02-12 02:19:13.610990605 +0100
@@ -22,6 +22,7 @@ int main(int argc, char **argv)
     // because we still need to do some cleanup.
     signal(SIGHUP, sighupHandler);
     qputenv("QT_NO_XDG_DESKTOP_PORTAL", QByteArrayView("1"));
+    qputenv("QT_QPA_PLATFORM", "xcb");
 
     QCoreApplication app(argc, argv);
 
